Sources/Afluent/Workers/SingleValueSubject.swift

```diff
@@ -30,19 +30,28 @@ public final class SingleValueSubject<Success: Sendable>: AsynchronousUnitOfWork
         AsynchronousOperation { [weak self] in
             guard let self else { throw CancellationError() }
 
-            self.lock()
-            if case .sentValue(let success) = self.subjectState {
-                self.unlock()
-                return success
-            } else if case .sentError(let error) = self.subjectState {
-                self.unlock()
-                throw error
+            func getSentValue() throws -> Success? {
+                try _lock.protect {
+                    if case .sentValue(let success) = self.subjectState {
+                        return success
+                    } else if case .sentError(let error) = self.subjectState {
+                        throw error
+                    }
+                    return nil
+                }
             }
-            self.unlock()
+
+            if let success = try getSentValue() { return success }
+
             return try await withUnsafeThrowingContinuation { continuation in
-                self.lock()
-                self.subjectState = .hasContinuation(continuation)
-                self.unlock()
+                self._lock.protect {
+                    do {
+                        if let success = try getSentValue() { continuation.resume(returning: success) }
```

Comment:
Hehe, sorry. I don't know if it's safe to call the continuation while locked, can you call the continuation outside of locks? It actually looks as though you only need to protect the assignment of `subjectState` since `getSentValue()` is protected.